test/test_scripts_run_task.py

```diff
@@ -346,6 +462,7 @@ def test_git_checkout(
             commit=None,
             ssh_key_file=None,
             ssh_known_hosts_file=None,
+            submodules=submodules,
         )
 
         # Check desired files exist
```

Comment:
There's an assert further down to check whether the outer repo is at the right revision. There's an assert checking for expected files makes sure _something_ was checked out for the submodules, which I guess is good enough here, since the submodules themselves control which revs are used?

Reply:
That is correct, a submodule in the superproject specifies the exact hash of child project to be checked out, and the `git submodule update --force` flag will ensure that the revision specified by the superproject is checked out.